### ISSUE TITLE
v2: selfReporting generates incomplete metric grafana_kubernetes_monitoring_build_info

### ISSUE THREAD

**Issue by marcomusso** [grafana/k8s-monitoring-helm#875]
From a vanilla values file I enabled selfReporting to ensure all metrics checks are green in Grafana Cloud (Home => Infrastructure => Kubernetes => Configuration => Metrics status).

I defined 3 destinations (prometheus, loki and otlp) and deployed.

In the resulting alloy configs only the singleton shows the reporting blocks added by the chart templates (which is probably correct as it's the [$chosenCollector](https://github.com/grafana/k8s-monitoring-helm/blob/6d61defc346765f7e581b06ff39a2ab13013fe4b/charts/k8s-monitoring/templates/features/_feature_self_reporting.tpl#L10) even if I would expect the receiver to be higher priority if the "list" is ordered) and specifically generates this metric:

```
# TYPE grafana_kubernetes_monitoring_build_info gauge                                                        grafana_kubernetes_monitoring_build_info{version="2.0.0-rc.2", namespace="grafana-k8s-monitoring-v2", platform=""} 1
```

which lacks the `cluster` label (ie it's not added even in further relabeling blocks) thus failing the check:

![Image](https://github.com/user-attachments/assets/c8d298de-8484-4e31-88b6-75def3297d99)

Here we can query for that metric and see its labels:

![Image](https://github.com/user-attachments/assets/be037a4b-03f7-4d1c-9ba3-67e88c11c875)

Please note: the `cluster` label is missing from all `grafana_kubernetes.*` metrics, is that expected/correct?

As a side note: the `selfReporting.scrapeInterval` key can be set but it's overridden by the global one so I don't see the point in setting it (and the comment/description in the values file doesn't help understand how to use it).

**Comment by marcomusso** [comment]
Fact is: I set 5m in that scrape interval and yet I god samples each minute that's why I said it was overridden (maybe poor choice of word, I didn't investigate too much but trusted the comment in the value file).

**Comment by marcomusso** [comment]
btw: the cluster label is now present in all `grafana_kubernetes*` metrics. Probably before it was not added as "external label" because it was [not sent to a prometheus destination](https://github.com/grafana/k8s-monitoring-helm/pull/878/files#diff-62c7fc838588ca15c3a6a0f020f1d2bb98cc52deaa5e310599ae90db6b8eb80fL26) (if I read that line correctly)?

**Comment by petewall** [comment]
Yeah, I just fixed an issue where it would request prometheus-ecosystem metrics destinations, but then try to use otlp-ecosystem metrics destinations. I fixed it to be consistent (prefer prometheus ecosystem).

I also need to fix the otlp destination to set `cluster` as well as `k8s.cluster.name`, which matches the behavior of loki and prometheus destinations.
